### PR TITLE
feat: add file existence check example

### PR DESCRIPTION
**Pull request by lino-levan**
95% of this example is quite literally text, but I think this makes sense given the circumstances.

### CODE REVIEW

File: data/checking-file-existence.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// Let's say we wanted to create a folder if one doesn't
+// already exist. Logically it makes sense to first verify
+// that the folder exists, then try to create it right?
+// Wrong. This will create a race condition where if a folder
+// gets created in between when you check if the folder exists
+// and when you create a folder, your program will crash.
+// Instead, you should just create a folder and try to catch
+// errors like so.
+try {
+  Deno.mkdir("new_dir");
```
Comment (kt3k) [inline data/checking-file-existence.ts:21]:
```suggestion
  await Deno.mkdir("new_dir");
```

File: data/checking-file-existence.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    throw err;
+  }
+}
+
+// This applies to almost every usecase. If you have a niche
+// usecase that requires you to check for existence of a file
+// without doing an filesystem operations other than that
+// (which is quite rare), the standard library provides a
+// utiltity for this.
+import { exists } from "$std/fs/exists.ts";
```
Comment (kt3k) [inline data/checking-file-existence.ts:33]: Are we settled on recovering/undeprecating `fs/exists`?

I think it's safer to suggest `try { await fs.lstat(...) } catch { ... }` for checking file existence.

Comment (lino-levan) [reply data/checking-file-existence.ts:33]: Yeah I wasn't sure what to do on this one and did it this way to open a discussion. 

If we decide to un-deprecate `fs/exists`, then that advice becomes a little dubious. If we do manage to remove `fs/exists` (which feels like it'll be very hard), then the way this is currently written will have to be rewritten. I'll rewrite it for now to use lstat. If we un-deprecate `fs/exists/` I'll look into changing this example.

I think the current understanding is `fs/exists` is permanently deprecated but is unlikely to be removed.

**Review by kt3k — APPROVED** [review]
LGTM. Thanks for updating!

**Pull request merged**